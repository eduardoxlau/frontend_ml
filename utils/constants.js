export const api = "https://limitless-atoll-23313.herokuapp.com/api/items";
//export const api = "http://localhost:4000/api/items";
export const ROUTECHANGESTART = "routeChangeStart";
export const ROUTECHANGECOMPLETE = "routeChangeComplete";

export const Items = [
  {
    id: "MLA848706209",
    title: "iPhone 11 64 Gb Blanco",
    price: {
      currency: "ARS",
      amount: 153000,
      decimals: 0,
    },
    picture: "http://http2.mlstatic.com/D_641268-MLA43654417348_102020-I.jpg",
    condition: "new",
    free_shipping: true,
    city: "Barracas",
  },
  {
    id: "MLA851450719",
    title: "iPhone 7 32 Gb Negro Mate",
    price: {
      currency: "ARS",
      amount: 76131,
      decimals: 0,
    },
    picture: "http://http2.mlstatic.com/D_942036-MLA31003118587_062019-I.jpg",
    condition: "new",
    free_shipping: true,
    city: "Palermo",
  },
];

export const Item = {
  id: "MLA771862889",
  title: "18 X $ 6143.- Motomel Cg 150 S2 0km Cycles",
  price: {
    currency: "ARS",
    amount: 89900,
    decimals: 0,
  },
  picture: "http://http2.mlstatic.com/D_670368-MLA25652564579_062017-O.jpg",
  condition: "new",
  sold_quantity: 0,
  free_shipping: false,
  description:
    "MOTOMEL CG 150 S2\r\n\r\nTENELA CON UN ANTICIPO Y 12, 18 CUOTAS CON TARJETA DE CRÃ‰DITO\r\n\r\nFINANCIALA 100% HASTA 18 CUOTAS FIJAS Y EN PESOS, CON TARJETA VISA O MASTER BANCARIA\r\nLLAMANOS AL (011) 5219-1111\r\n\r\nSU CONSULTA NO MOLESTA! CYCLES MOTOSHOP",
  sold: 0,
};
