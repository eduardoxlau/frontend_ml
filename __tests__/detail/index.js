import { shallow } from "enzyme";
import { Item as dataDefault } from "utils/constants";
import Item from "../../pages/items/[id].js";

const mockITEM = dataDefault;
jest.mock("../../fetch_api", () => ({
  getItem: () => Promise.resolve({ item: mockITEM }),
}));

describe("Items", () => {
  it("should render correctly", () => {
    const wrapper = shallow(<Item />);
    expect(wrapper).toMatchSnapshot();
  });

  it("shouldn return one object", async () => {
    const props = await Item.getInitialProps({
      query: { id: "1" },
    });
    expect(props.item).toEqual(dataDefault);
  });
});
