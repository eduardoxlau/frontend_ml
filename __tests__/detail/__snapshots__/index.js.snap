// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Items should render correctly 1`] = `
<Fragment>
  <BreadcrumbCustom />
  <container__Container
    mb={40}
    mt={-20}
  >
    <Detail
      loading={false}
    />
  </container__Container>
</Fragment>
`;
